examples/sampling/pyg/node_classification.py

```diff
@@ -0,0 +1,184 @@
+main: 
+
+main
+│
+├───> Load and preprocess dataset (GraphBolt)
+│     │
+│     └───> Utilize GraphBolt's BuiltinDataset for dataset handling
+│
+├───> Instantiate the SAGE model (PyTorch Geometric)
+│     │
+│     └───> Define the GraphSAGE model architecture
+│
+├───> Train the model
+│     │
+│     ├───> Mini-Batch Processing with GraphBolt
+│     │     │
+│     │     └───> Efficient handling of mini-batches using GraphBolt's utilities
+│     │
+│     └───> Training Loop
+│           │
+│           ├───> Forward and backward passes
+│           │
+│           └───> Parameters optimization
+│
+└───> Evaluate the model
+      │
+      └───> Performance assessment on validation and test datasets
+            │
+            └───> Accuracy and other relevant metrics calculation
+
+
+"""
+
+
+
+
+import torch
+import torch.nn.functional as F
+from torch_geometric.nn import SAGEConv
+from dgl.graphbolt import BuiltinDataset, DataLoader, ItemSampler, FeatureFetcher, CopyTo
+
+
+
+
+class GraphSAGE(torch.nn.Module):
+    #####################################################################
+    # (HIGHLIGHT) Define the GraphSAGE model architecture.
+    # 
+    # - This class inherits from `torch.nn.Module`.
+    # - Two convolutional layers are created using the SAGEConv class from PyG.
+    # - 'in_size', 'hidden_size', 'out_size' are the sizes of
+    #   the input, hidden, and output features, respectively.
+    # - The forward method defines the computation performed at every call.
+    #####################################################################
+    def __init__(self, in_size, hidden_size, out_size):
+        super(GraphSAGE, self).__init__()
+        self.conv1 = SAGEConv(in_size, hidden_size)
+        self.conv2 = SAGEConv(hidden_size, out_size)
+
+    def forward(self, blocks, x, device):
+        h = x
+        for i, block in enumerate(blocks):
+            edge_index = block.edges()
+            edge_index = torch.stack([edge_index[0], edge_index[1]], dim=0).to(device)
+            h = self.conv1(h, edge_index) if i == 0 else self.conv2(h, edge_index)
+            if i != len(blocks) - 1:
+                h = F.relu(h)
+            h = h[:block.number_of_dst_nodes()]
+        return h
+
+
+def create_dataloader(dataset_set, graph, feature, device, shuffle):
+    #####################################################################
+    # (HIGHLIGHT) Create a data loader for efficiently loading graph data.
+    #
+    # - 'ItemSampler' samples mini-batches of node IDs from the dataset.
+    # - 'sample_neighbor' performs neighbor sampling on the graph.
+    # - 'FeatureFetcher' fetches node features based on the sampled subgraph.
+    # - 'CopyTo' copies the fetched data to the specified device.
+    #####################################################################
+    datapipe = ItemSampler(dataset_set, batch_size=1024, shuffle=shuffle)
+    datapipe = datapipe.sample_neighbor(graph, [4, 4])
```

Comment:
```suggestion
    datapipe = datapipe.sample_neighbor(graph, [10, 10, 10])
```
@Rhett-Ying Do you think we can showcase `sample_layer_neighbor` here instead of `sample_neighbor` as PyG does not have it? This should make GraphBolt even more attractive to users.

Reply:
`sample_neighbor` is consistent with other counterpart in DGL. But we could add comment here that more samplers are available such as `layer`.